node-manager/src/dlcstorage.rs

```diff
@@ -0,0 +1,365 @@
+use crate::localstorage::MutinyBrowserStorage;
+use anyhow::Context;
+use bitcoin_hashes::hex::ToHex;
+use dlc_manager::chain_monitor::ChainMonitor;
+use dlc_manager::channel::accepted_channel::AcceptedChannel;
+use dlc_manager::channel::offered_channel::OfferedChannel;
+use dlc_manager::channel::signed_channel::{SignedChannel, SignedChannelStateType};
+use dlc_manager::channel::{Channel, FailedAccept, FailedSign};
+use dlc_manager::contract::accepted_contract::AcceptedContract;
+use dlc_manager::contract::offered_contract::OfferedContract;
+use dlc_manager::contract::ser::Serializable;
+use dlc_manager::contract::signed_contract::SignedContract;
+use dlc_manager::contract::{
+    ClosedContract, Contract, FailedAcceptContract, FailedSignContract, PreClosedContract,
+};
+use dlc_manager::{error::Error, ChannelId, ContractId, Storage};
+use std::collections::HashMap;
+use std::convert::TryInto;
+use std::io::{Cursor, Read};
+
+const DLC_CONTRACT_KEY_PREFIX: &str = "dlc_contract_";
+const DLC_CHANNEL_KEY_PREFIX: &str = "dlc_channel_";
+const DLC_CHAIN_MONITOR_KEY: &str = "dlc_chain_monitor";
+
+impl Storage for MutinyBrowserStorage {
+    fn get_contract(&self, id: &ContractId) -> Result<Option<Contract>, Error> {
+        let key = format! {"{DLC_CONTRACT_KEY_PREFIX}{}", id.to_hex()};
+        let result: Result<Vec<u8>, MutinyStorageError> = self.get(key);
+        match result {
+            Ok(contract_bytes) => {
+                let contract = deserialize_contract(contract_bytes)
+                    .map_err(|e| Error::StorageError(format!("{}", e)))?;
+                Ok(Some(contract))
+            }
+            Err(_) => Ok(None),
+        }
+    }
+
+    fn get_contracts(&self) -> Result<Vec<Contract>, Error> {
+        let map: HashMap<String, Vec<u8>> = self.scan(DLC_CONTRACT_KEY_PREFIX, None);
+        let contracts = map.into_values().map(|b| deserialize_contract(b).unwrap());
+        Ok(contracts.collect())
+    }
+
+    fn create_contract(&mut self, contract: &OfferedContract) -> Result<(), Error> {
+        let serialized = serialize_contract(&Contract::Offered(contract.clone()))?;
+        let id = contract.id.to_hex();
+        let key = format! {"{DLC_CONTRACT_KEY_PREFIX}{id}"};
+
+        self.set(key, serialized)?;
+        Ok(())
+    }
+
+    fn delete_contract(&mut self, id: &ContractId) -> Result<(), Error> {
+        let key = format! {"{DLC_CONTRACT_KEY_PREFIX}{}", id.to_hex()};
+        MutinyBrowserStorage::delete(key);
+        Ok(())
+    }
+
+    fn update_contract(&mut self, contract: &Contract) -> Result<(), Error> {
+        let serialized = serialize_contract(contract)?;
+        match contract {
+            a @ Contract::Accepted(_) | a @ Contract::Signed(_) => {
+                let key = format! {"{DLC_CONTRACT_KEY_PREFIX}{}", a.get_temporary_id().to_hex()};
+                MutinyBrowserStorage::delete(key);
+            }
+            _ => {}
+        };
```

Comment:
Should you delete after `set` instead? What if there's a problem saving the contract?